awscli/customizations/s3/fileinfo.py

```diff
@@ -266,7 +270,22 @@ def _handle_upload(self, body):
 
     def _inject_content_type(self, params, filename):
         # Add a content type param if we can guess the type.
-        guessed_type = guess_content_type(filename)
+        guessed_type = None
+        try:
+            guessed_type = guess_content_type(filename)
+        # This catches a bug in the mimetype libary where some MIME types
+        # specifically on windows machines cause a UnicodeDecodeError
+        # because the MIME type in the Windows registery has an encoding
+        # that cannot be properly encoded using the default system encoding.
+        # https://bugs.python.org/issue9291
+        #
+        # So instead of hard failing, just log the issue and fall back to the
+        # default guessed content type of None.
+        except UnicodeDecodeError:
+            LOGGER.debug(
+                'Unable to guess content type for %s due to '
+                'UnicodeDecodeError: ', filename, exc_info=True
+            )
         if guessed_type is not None:
```

Comment:
I think this should be moved in the try block before the except block. We only need to check if `guessed_type` is not None when a unicode exception was not raised.